src/video/sdl_v.cpp

```diff
@@ -155,19 +164,14 @@ static void DrawSurfaceToScreen()
 	if (n == 0) return;
 
 	_num_dirty_rects = 0;
-	if (n > MAX_DIRTY_RECTS) {
-		if (_sdl_screen != _sdl_realscreen) {
-			SDL_CALL SDL_BlitSurface(_sdl_screen, NULL, _sdl_realscreen, NULL);
-		}
-		SDL_CALL SDL_UpdateRect(_sdl_realscreen, 0, 0, 0, 0);
-	} else {
-		if (_sdl_screen != _sdl_realscreen) {
-			for (int i = 0; i < n; i++) {
-				SDL_CALL SDL_BlitSurface(_sdl_screen, &_dirty_rects[i], _sdl_realscreen, &_dirty_rects[i]);
-			}
-		}
-		SDL_CALL SDL_UpdateRects(_sdl_realscreen, n, _dirty_rects);
-	}
+
+	// Just update the entire screen. Updating only the _dirty_rects
+	// seems to cause screen flickering, but that's how the SDL 1.2
+	// code worked.
+	SDL_CALL SDL_UpdateTexture(_sdl_texture, NULL, _sdl_screen->pixels, _sdl_screen->pitch);
+	SDL_CALL SDL_RenderClear(_sdl_renderer);
+	SDL_CALL SDL_RenderCopy(_sdl_renderer, _sdl_texture, NULL, NULL);
+	SDL_CALL SDL_RenderPresent(_sdl_renderer);
```

Comment:
Updating just dirty rects was a _major_ performance improvement that definitely used to work.

I wonder why it stopped working? Our code didn't change, but OSes have... (It stopped working for the Windows driver too!)

Reply:
Interesting, the [MigrationGuide](https://wiki.libsdl.org/MigrationGuide) mentions this here:

> At the end of the frame, we want to upload to the texture like this:
> 
>   SDL_UpdateTexture(sdlTexture, NULL, myPixels, 640 * sizeof (Uint32));
> 
> This will upload your pixels to GPU memory. That NULL can be a subregion if you want to mess around with dirty rectangles, but chances are modern hardware can just swallow the whole framebuffer without much trouble. The final argument is the pitch--the number of bytes from the start of one row to the next--and since we have a linear RGBA buffer in this example, it's just 640 times 4 (r,g,b,a).

When I've adapted `DrawSurfaceToScreen()` to use the dirty rectangles system, everything works, but you can actually see the dirty rectangles getting re-drawn and there is some screen flickering. There may be a way to get this working as maybe I was doing something wrong, I just haven't spent too much time on this because I wasn't sure it was necessary.

Reply:
Yeah, maybe it's fine, 10+ years later? Also the GUI timer changes alleviate the load a bit.